trinity/extensibility/plugin_manager.py

```diff
@@ -116,65 +187,64 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
 
             plugin._start()
             self._started_plugins.append(plugin)
-            self._logger.info("Plugin started: {}".format(plugin.name))
+            self._logger.info("Plugin started: %s", plugin.name)
             self.broadcast(PluginStartedEvent(plugin), plugin)
 
     def prepare(self,
                 args: Namespace,
                 chain_config: ChainConfig,
                 boot_kwargs: Dict[str, Any] = None) -> None:
+        """
+        Create a ``PluginContext`` for every plugin that this plugin manager instance
+        is responsible for.
+        """
         for plugin in self._plugin_store:
 
-            if not self._is_responsible_for_plugin(plugin):
+            if not self._scope.is_responsible_for_plugin(plugin):
                 continue
 
-            context = self._create_context_for_plugin(plugin, args, chain_config, boot_kwargs)
+            context = self._scope.create_plugin_context(plugin, args, chain_config, boot_kwargs)
             plugin.set_context(context)
 
-    def shutdown(self) -> None:
+    def shutdown_nowait(self) -> None:
```

Comment:
*minor naming nitpick*

I'm not sure the `_nowait` suffix is being used here in quite the same way as other `asyncio` library usages. This is potentially a long blocking wait for each plugin to shut down, right? Seems like it's more like a `shutdown_blocking` to me.

Reply:
Ah, *that* seems like the right name for it :+1: